agent/xds/listeners.go

```diff
@@ -218,26 +218,33 @@ func (s *ResourceGenerator) listenersFromSnapshotConnectProxy(cfgSnap *proxycfg.
 		// as opposed to via a virtual IP.
 		var passthroughChains []*envoy_listener_v3.FilterChain
 
-		for uid, passthrough := range cfgSnap.ConnectProxy.PassthroughUpstreams {
-			u := structs.Upstream{
-				DestinationName:      uid.Name,
-				DestinationNamespace: uid.NamespaceOrDefault(),
-				DestinationPartition: uid.PartitionOrDefault(),
-			}
+		for _, target := range cfgSnap.ConnectProxy.PassthroughUpstreams {
+			for tid, addrs := range target {
+				uid := proxycfg.NewUpstreamIDFromTargetID(tid)
 
-			filterName := fmt.Sprintf("%s.%s.%s.%s", u.DestinationName, u.DestinationNamespace, u.DestinationPartition, cfgSnap.Datacenter)
+				sni := connect.ServiceSNI(
+					uid.Name, "", uid.NamespaceOrDefault(), uid.PartitionOrDefault(), cfgSnap.Datacenter, cfgSnap.Roots.TrustDomain)
 
-			filterChain, err := s.makeUpstreamFilterChain(filterChainOpts{
-				clusterName: "passthrough~" + passthrough.SNI,
-				filterName:  filterName,
-				protocol:    "tcp",
-			})
-			if err != nil {
-				return nil, err
-			}
-			filterChain.FilterChainMatch = makeFilterChainMatchFromAddrs(passthrough.Addrs)
+				u := structs.Upstream{
```

Comment:
I don't think you even need to make this object, since you only use it to acces the same fields on line 234 that you used to construct it.